### DISCUSSION TITLE
Update table from .csv

### DISCUSSION THREAD

**Discussion by phoskee** [zauberzeug/nicegui#408] [Q&A] [answered]
I managed to create the table from a .csv file, however, I am unable to update it as the data in the .csv file changes.
I can only use `.append()` and this creates duplicates every time I call the function (which is expected, since I am not deleting the rows). Any advice?

```py
def popula_table():
    a = []
    with open(filename, 'rt') as f:
        reader = csv.DictReader(f)
        for row in reader:
            table_accessi.options['rowData'].append(row)
    table_accessi.update()
```

**Comment by AL-THE-BOT-FATHER** [comment, marked as answer]
In your code your for loop should be appending to the "a" list. 
Then you would do: 
```py
table_accessi.options['rowData'] = a
table_accessi.update()
```
Make sure the data formatted as a list of dictionaries with the keys matching the ones in your table. 
```py
table_data = [
    {'name': 'Alice', 'age': 58},
    {'name': 'Bob', 'age': 37},
    {'name': 'Charlie', 'age': 23}]
```

**Reply by phoskee** [reply]
Thanks a lot for the quick reply!

It was so simple, I fixed the code with your advice:

```py
def popula_table():
    a = []
    with open(filename, 'rt') as f:
        reader = csv.DictReader(f)
        for row in reader:
            a.append(row)
    table_accessi.options['rowData'] = a
    table_accessi.update()
```

I am very grateful to you, thank you for the work you are doing!

**Reply by AL-THE-BOT-FATHER** [reply]
This isn't my project. I just knew the answer to your question. Cheers.

**Reply by linkwww8** [reply]
@phoskee @Allen-Taylor Thanks for the help code above! 

I stuck on how to stop table resetting view to top on every update, do you know way to scroll to bottom/newest row each time? Thxs!
